spec/controllers/posts_controller_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  let(:user) { User.create!(name: 'Test User', posts_counter: 0) }
+
+  describe 'GET #index' do
+    it 'assigns the requested user as @user' do
+      get :index, params: { user_id: user.to_param }
+      expect(assigns(:user)).to eq(user)
+    end
+
+    it 'assigns the user posts as @posts' do
+      post = user.posts.create!(title: 'Test Post', likes_counter: 0, comments_counter: 0)
+      get :index, params: { user_id: user.to_param }
+      expect(assigns(:posts)).to eq([post])
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when the post exists' do
+      it 'assigns the requested user as @user' do
+        post = user.posts.create!(title: 'Test Post', likes_counter: 0, comments_counter: 0)
+        get :show, params: { user_id: user.to_param, id: post.to_param }
+        expect(assigns(:user)).to eq(user)
+      end
+
+      it 'assigns the requested post as @post' do
+        post = user.posts.create!(title: 'Test Post', likes_counter: 0, comments_counter: 0)
+        get :show, params: { user_id: user.to_param, id: post.to_param }
+        expect(assigns(:post)).to eq(post)
+      end
+    end
+
+    context 'when the post does not exist' do
+      it 'redirects to the user posts list' do
+        get :show, params: { user_id: user.to_param, id: -1 }
+        expect(response).to redirect_to(user_posts_path(user))
+      end
+    end
+  end
```

Comment:
| Project Requirements 📝 |
|-|
| ![image](https://github.com/fuadmoin/Blog-app/assets/93039359/601cfa21-c099-44f7-88c1-39b092274cfb) |


- [ ] It would be nice if you follow the project requirements because in the project requirements, it is clearly mentioned that you have to check the things shown in the image above ⬆️ 👍

Hint: You can use the following matchers to complete the testing according to the project requirements 👍

```ruby

--- These below are just examples so you need to write your tests in your own way

RSpec.describe 'Posts', type: :request do
  describe 'GET /index' do
    before(:example) { get '/users/1/posts' }
    it 'renders http status' do
      expect(response).to have_http_status :ok
    end

    it "renders 'index' template" do
      expect(response).to render_template :index
    end

    it 'renders correct placeholder' do
      expect(response.body).to include '<h1>Posts#index</h1>'
    end
  end

  describe 'GET /show' do
    before(:example) { get '/users/1/posts/1' }
    it 'renders http status' do
      expect(response).to have_http_status :ok
    end

    it "renders 'show' template" do
      expect(response).to render_template :show
    end

    it 'renders correct placeholder' do
      expect(response.body).to include '<h1>Single post</h1>'
    end
  end
end
```